bundler/lib/bundler/compact_index_client/updater.rb

```diff
@@ -53,13 +53,13 @@ def replace(remote_path, local_path, etag_path)
         response = @fetcher.call(remote_path, request_headers(etag))
         return true if response.is_a?(Gem::Net::HTTPNotModified)
         CacheFile.write(local_path, response.body, parse_digests(response))
-        CacheFile.write(etag_path, etag(response))
+        CacheFile.write(etag_path, etag_from_response(response))
       end
 
       def request_headers(etag, range_start = nil)
         headers = {}
         headers["Range"] = "bytes=#{range_start}-" if range_start
-        headers["If-None-Match"] = etag if etag
+        headers["If-None-Match"] = %("#{etag}") if etag && !etag.empty?
```

Comment:
In which case the passed etag will be an empty string?

Reply:
Guarding against an empty file on write error 🤷‍♂️ Maybe there's no harm in sending "" and we can avoid the check in that rare case.

Reply:
I removed the empty guard. It should have no negative effect, and it should almost never happen.